src/libraries/System.Private.CoreLib/src/System/Half.cs

```diff
@@ -0,0 +1,707 @@
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Globalization;
+using System.Numerics;
+using System.Runtime.InteropServices;
+
+namespace System
+{
+    // Portions of the code implemented below are based on the 'Berkeley SoftFloat Release 3e' algorithms.
+
+    /// <summary>
+    /// An IEEE 754 compliant float16 type.
+    /// </summary>
+    [StructLayout(LayoutKind.Sequential)]
+    public readonly struct Half : IComparable, IFormattable, IComparable<Half>, IEquatable<Half>, ISpanFormattable
+    {
+        private const NumberStyles DefaultParseStyle = NumberStyles.Float | NumberStyles.AllowThousands;
+
+        // Constants for manipulating the private bit-representation
+
+        private const ushort SignMask = 0x8000;
+        private const ushort SignShift = 15;
+
+        private const ushort ExponentMask = 0x7C00;
+        private const ushort ExponentShift = 10;
+
+        private const ushort SignificandMask = 0x03FF;
+        private const ushort SignificandShift = 0;
+
+        private const ushort MinSign = 0;
+        private const ushort MaxSign = 1;
+
+        private const ushort MinExponent = 0x00;
+        private const ushort MaxExponent = 0x1F;
+
+        private const ushort MinSignificand = 0x0000;
+        private const ushort MaxSignificand = 0x03FF;
+
+        // Constants representing the private bit-representation for various default values
+
+        private const ushort PositiveZeroBits = 0x0000;
+        private const ushort NegativeZeroBits = 0x8000;
+
+        private const ushort EpsilonBits = 0x0001;
+
+        private const ushort PositiveInfinityBits = 0x7C00;
+        private const ushort NegativeInfinityBits = 0xFC00;
+
+        private const ushort PositiveQNaNBits = 0x7E00;
+        private const ushort NegativeQNaNBits = 0xFE00;
+
+        private const ushort MinValueBits = 0xFBFF;
+        private const ushort MaxValueBits = 0x7BFF;
+
+
+        // Well-defined and commonly used values
+
+        private static readonly Half s_epsilon = new Half(EpsilonBits);                        //  5.9604645E-08
+        public static Half Epsilon => s_epsilon;
+
+        private static readonly Half s_positiveInfinity = new Half(PositiveInfinityBits);      //  1.0 / 0.0
+        public static Half PositiveInfinity => s_positiveInfinity;
+
+        private static readonly Half s_negativeInfinity = new Half(NegativeInfinityBits);      // -1.0 / 0.0
+        public static Half NegativeInfinity => s_negativeInfinity;
+
+        private static readonly Half s_naN = new Half(NegativeQNaNBits);                       //  0.0 / 0.0
+        public static Half NaN => s_naN;
+
+        private static readonly Half _MinValue = new Half(MinValueBits);                      // -65504
+        public static Half MinValue => _MinValue;
```

Comment:
```suggestion
        public static Half MinValue =>  new Half(MinValueBits);
```
Caching the trivial value like in a static field is deoptimization. It should be returned directly.
Same for all other cases.